pandas/indexes/numeric.py

```diff
@@ -22,6 +22,36 @@ class NumericIndex(Index):
     """
     _is_numeric_dtype = True
 
+    def __new__(cls, data=None, dtype=None, copy=False, name=None,
+                fastpath=False):
+
+        if fastpath:
+            return cls._simple_new(data, name=name)
+
+        if dtype and np.dtype(dtype) != cls._default_dtype:
```

Comment:
I get the point, but not your code. If it's as you say, we won't look at the ``dtype`` argument _at all_, since we fall back to ``cls._default_dtype`` anyway. We just look at the data dtype.